command/connect/envoy/envoy.go

```diff
@@ -321,6 +331,27 @@ func (c *cmd) lookupProxyIDForSidecar() (string, error) {
 	return proxyCmd.LookupProxyIDForSidecar(c.client, c.sidecarFor)
 }
 
+func (c *cmd) lookupGRPCPort() (int, error) {
+	self, err := c.client.Agent().Self()
+	if err != nil {
+		return 0, err
+	}
+	cfg, ok := self["DebugConfig"] // TODO: use string?
+	if !ok {
+		return 0, fmt.Errorf("no debug config") // TODO: error msg: what is acceptable to show to client?
```

Comment:
I'd suggest something simple. I _think_ this could only happen due to a bug or being run against a different version of Consul that happens not to export that field right? It's not optional to expose that?

If correct it's not likely to come up but something like:
```suggestion
		return 0, fmt.Errorf("unexpected agent response: no debug config")
```
Seems fine to me

Reply:
awesome!! thanks for these error suggestions